buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/PerformanceTest.java

```diff
@@ -79,44 +132,82 @@ public String getRuns() {
-    }
-
     @Option(option = "channel", description = "Channel to use when running the performance test. By default, 'commits'.")
     public void setChannel(@Nullable String channel) {
         this.channel = channel;
-        systemProperty("org.gradle.performance.execution.channel", channel);
     }
 
-    @OutputDirectory
-    public File getDebugArtifactsDirectory() {
-        return debugArtifactsDirectory;
+    @Nullable @Optional @Input
+    public String getChannel() {
+        return channel;
     }
 
-    public void setDebugArtifactsDirectory(File debugArtifactsDirectory) {
-        this.debugArtifactsDirectory = debugArtifactsDirectory;
+    public void setResultStoreClass(String resultStoreClass) {
+        this.resultStoreClass = resultStoreClass;
+    }
+
+    @Nullable @Optional @Input
+    public String getResultStoreClass() {
+        return resultStoreClass;
     }
 
     @Option(option = "flamegraphs", description = "If set to 'true', activates flamegraphs and stores them into the 'flames' directory name under the debug artifacts directory.")
     public void setFlamegraphs(String flamegraphs) {
-        if ("true".equals(flamegraphs)) {
+        this.flamegraphs = "true".equals(flamegraphs);
+    }
+
+    @Nullable @Optional @Input
+    public Boolean isFlamegraphs() {
+        return flamegraphs;
+    }
+
+    @Input
+    public String getDatabaseUrl() {
+        return databaseParameters.get("org.gradle.performance.db.url");
+    }
+
+    public void addDatabaseParameters(Map<String, String> databaseConnectionParameters) {
+        this.databaseParameters.putAll(databaseConnectionParameters);
+    }
+
+    private class PerformanceTestJvmArgumentsProvider implements CommandLineArgumentProvider {
+        @Override
+        public Iterable<String> asArguments() {
+            List<String> result = new ArrayList<>();
+            addExecutionParameters(result);
+            addDatabaseParameters(result);
+            return result;
+        }
+
+        private void addExecutionParameters(List<String> result) {
+            addSystemPropertyIfExist(result, "org.gradle.performance.scenarios", scenarios);
+            addSystemPropertyIfExist(result, "org.gradle.performance.baselines", baselines);
+            addSystemPropertyIfExist(result, "org.gradle.performance.execution.warmups", warmups);
+            addSystemPropertyIfExist(result, "org.gradle.performance.execution.runs", runs);
+            addSystemPropertyIfExist(result, "org.gradle.performance.execution.checks", checks);
+            addSystemPropertyIfExist(result, "org.gradle.performance.execution.channel", channel);
+
             File artifactsDirectory = new File(getDebugArtifactsDirectory(), "flames");
-            systemProperty("org.gradle.performance.flameGraphTargetDir", artifactsDirectory.getAbsolutePath());
+            addSystemPropertyIfExist(result, "org.gradle.performance.flameGraphTargetDir", artifactsDirectory.getAbsolutePath());
```

Comment:
I guess we only want to add that directory if `flamegraphs` is true, right?

Reply:
Right.